velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -369,5 +383,38 @@ TEST_F(StringTest, rtrim) {
       "\u6574\u6570 \u6570\u636E!");
 }
 
+TEST_F(StringTest, substring) {
+  EXPECT_EQ(substring("example", 0, 2), "ex");
+  EXPECT_EQ(substring("example", 1, -1), "");
+  EXPECT_EQ(substring("example", 1, 0), "");
+  EXPECT_EQ(substring("example", 1, 2), "ex");
+  EXPECT_EQ(substring("example", 1, 7), "example");
+  EXPECT_EQ(substring("example", 1, 100), "example");
+  EXPECT_EQ(substring("example", 2, 2), "xa");
+  EXPECT_EQ(substring("example", 8, 2), "");
+  EXPECT_EQ(substring("example", -2, 2), "le");
+  EXPECT_EQ(substring("example", -7, 2), "ex");
+  EXPECT_EQ(substring("example", -8, 2), "e");
+  EXPECT_EQ(substring("example", -9, 2), "");
```

Comment:
I also noted the issue in `substr` for spark sql recently. Can you add the below two test cases? Looks representative. 
```
EXPECT_EQ(substring("example", -7, 7), "example");
EXPECT_EQ(substring("example", -9, 9), "example");
```

Reply:
@PHILO-HE Thanks!

Done.